src/typed-method-types/workflows/triggers/mod.ts

```diff
@@ -11,64 +28,122 @@ export const TriggerTypes = {
+    : `${string}#/workflows/${string}`;
 
-export type BaseTrigger = {
+export type BaseTrigger<WorkflowDefinition extends WorkflowSchema> = {
   /** @description The type of trigger */
-  type: typeof TriggerTypes[keyof typeof TriggerTypes];
+  type: string;
   /** @description The workflow that the trigger initiates */
-  workflow: WorkflowStringFormat;
-  /** @description The inputs provided to the workflow */
-  inputs?: Record<string, WorkflowInput>;
+  workflow: WorkflowStringFormat<WorkflowDefinition["callback_id"]>;
   /** @description The name of the trigger */
   name: string;
   /** @description The description of the trigger */
   description?: string;
-};
+  // deno-lint-ignore no-explicit-any
+  [otherOptions: string]: any;
+} & WorkflowInputs<WorkflowDefinition>;
 
-export type TriggerIdType = {
-  /** @description The id of a specified trigger */
-  trigger_id: string;
+export type WorkflowSchema = {
+  callback_id?: string;
+  description?: string;
+  input_parameters?: InputParameterSchema;
+  // deno-lint-ignore no-explicit-any
+  output_parameters?: Record<string, any>;
+  title: string;
 };
 
-// A helper to make sure inputs are passed. Required for automated triggers
-export type RequiredInputs = Required<Pick<BaseTrigger, "inputs">>;
+type ResponseTypes<
+  WorkflowDefinition extends WorkflowSchema,
+> =
+  & ShortcutTriggerResponse<WorkflowDefinition>
+  & EventTriggerResponse<WorkflowDefinition>
+  & ScheduledTriggerResponse<WorkflowDefinition>
+  & WebhookTriggerResponse<WorkflowDefinition>;
+
+/** @description Response object content for delete method */
+type DeleteResponse = {
+  ok: true;
+};
 
-export type ValidTriggerTypes =
-  | EventTrigger
-  | ScheduledTrigger
-  | ShortcutTrigger
-  | WebhookTrigger;
+type DeleteTriggerResponse = Promise<
+  DeleteResponse | FailedTriggerResponse
+>;
 
-type BaseTriggerResponse = Promise<BaseResponse>;
-type TriggerResponse = BaseTriggerResponse;
 type ListArgs = {
   /** @description Lists triggers only if they owned by the caller */
   is_owner?: boolean;
   /** @description Lists triggers only if they have been published */
   is_published?: boolean;
 };
 
+type ValidTriggerResponseObjects =
+  | ShortcutTriggerResponseObject<WorkflowSchema>
+  | EventTriggerResponseObject<WorkflowSchema>
+  | ScheduledTriggerResponseObject<WorkflowSchema>
+  | WebhookTriggerResponseObject<WorkflowSchema>;
+
+type ListResponse = {
+  ok: true;
+  triggers: ValidTriggerResponseObjects[];
+};
+
+type ListTriggerResponse = Promise<
+  ListResponse | FailedTriggerResponse
+>;
+
+export type FailedTriggerResponse = BaseResponse & {
+  ok: false;
+  /** @description no trigger is returned on a failed response */
+  trigger?: never;
+};
+
+type TriggerIdType = {
+  /** @description The id of a specified trigger */
+  trigger_id: string;
+};
+
+export type ValidTriggerTypes<
+  WorkflowDefinition extends WorkflowSchema = DEFAULT_WORKFLOW_TYPE,
```

Comment:
This default allows workflow generics to not be passed while still making sure that inputs aren't treated as not existing.